src/dependencies/dss-direct-deposit/D3MInit.sol

```diff
@@ -0,0 +1,326 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import "dss-interfaces/dss/DssAutoLineAbstract.sol";
+import "dss-interfaces/dss/IlkRegistryAbstract.sol";
+import "dss-interfaces/utils/WardsAbstract.sol";
+import "dss-interfaces/ERC/GemAbstract.sol";
+import { DssInstance } from "dss-test/MCD.sol";
+import { ScriptTools } from "dss-test/ScriptTools.sol";
+
+import { D3MInstance } from "./D3MInstance.sol";
+import { D3MCoreInstance } from "./D3MCoreInstance.sol";
+
+interface ID3MPool {
+    function redeemable() external view returns (address);
+}
+
+interface D3MAavePoolLike {
+    function hub() external view returns (address);
+    function dai() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function vat() external view returns (address);
+    function file(bytes32, address) external;
+    function adai() external view returns (address);
+    function stableDebt() external view returns (address);
+    function variableDebt() external view returns (address);
+}
+
+interface D3MAaveRateTargetPlanLike {
+    function rely(address) external;
+    function file(bytes32, uint256) external;
+    function adai() external view returns (address);
+    function stableDebt() external view returns (address);
+    function variableDebt() external view returns (address);
+    function tack() external view returns (address);
+    function adaiRevision() external view returns (uint256);
+}
+
+interface D3MAaveBufferPlanLike {
+    function file(bytes32, uint256) external;
+    function adai() external view returns (address);
+}
+
+interface ADaiLike {
+    function ATOKEN_REVISION() external view returns (uint256);
+}
+
+interface D3MCompoundPoolLike {
+    function hub() external view returns (address);
+    function dai() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function vat() external view returns (address);
+    function file(bytes32, address) external;
+    function cDai() external view returns (address);
+    function comptroller() external view returns (address);
+    function comp() external view returns (address);
+}
+
+interface D3MCompoundRateTargetPlanLike {
+    function rely(address) external;
+    function file(bytes32, uint256) external;
+    function tacks(address) external view returns (uint256);
+    function delegates(address) external view returns (uint256);
+    function cDai() external view returns (address);
+}
+
+interface CDaiLike {
+    function interestRateModel() external view returns (address);
+    function implementation() external view returns (address);
+}
```

Comment:
Compound interfaces not being used. Could be removed.

Reply:
Same as above.